src/components/calc.js

```diff
@@ -1,23 +1,57 @@
 import React from 'react';
 import Display from './calc/display';
 import ButtonsContainer from './calc/buttonsContainer';
+import calculate from './calc/logic/calculate';
 import './calc.css';
 
-// eslint-disable-next-line react/prefer-stateless-function
 class Calc extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      displayValue: '0',
+      total: null,
+      next: null,
+      operation: null,
     };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
   }
 
+  handleClick = (e) => {
+    this.updateState(e.target.dataset.name);
+  };
+
+  handleKeyDown = (e) => {
+    e.preventDefault();
+  };
+
+  updateState = (key) => {
+    const { total, next, operation } = calculate(this.state, key);
+    if (total || total === null) {
+      this.setState({
+        total,
+      });
+    }
+    if (next || next === null) {
+      this.setState({
+        next,
+      });
+    }
+    if (operation || operation === null) {
+      this.setState({
+        operation,
+      });
+    }
+  };
```

Comment:
-[ ] Please use the same design and color scheme as given in guidelines for calculator